src/Kriti/Eval.hs

```diff
@@ -74,7 +79,13 @@ isString _ = False
 runEval :: B.ByteString -> ValueExt -> [(T.Text, J.Value)] -> Either EvalError J.Value
 runEval src template source =
   let ctx = Compat.fromList source
-   in runReader (runExceptT (eval template)) (src, ctx)
+   in runReader (runExceptT (evalWith Map.empty template)) (src, ctx)
+
+runEvalWith :: B.ByteString -> ValueExt -> [(T.Text, J.Value)] -> [(T.Text, J.Value -> Either CustomFunctionError J.Value)] -> Either EvalError J.Value
+runEvalWith src template source functionLst =
+  let ctx = Compat.fromList source
+      funcMap = Map.fromList functionLst
```

Comment:
If a list is being used, then this could potentially check for function name conflicts.

Reply:
Using a hashmap would be a better choice.